### ISSUE TITLE
Consider to change/improve configfile discovery

### ISSUE THREAD

**Issue by netzego** [google/yamlfmt#123]
1. `$PWD/.yamlfmt`
2. `$HOME/.yamlfmt`
3. `$XDG_CONFIG_HOME/yamlfmt/config` or `.../yamlfmt/yamlfmt` (without a dot)
4. `/etc/yamlfmt/config`

The finding algo should search upwards from the current working dir. This helps to prevents spamming `.yamlfmt` files all over the repo dirs. Without this feature it is very hard to track/change settings.

Imho, dot files inside `$XDG_CONFIG_HOME` are not useful.

A system wide place eg. `etc` would be nice. 

Thx for considering and your work.

**Comment by netzego** [comment]
@braydonk sure:
1) `$XDG_CONFIG_HOME` is a dotfile/folder itself. so lets take a look at for e.g. `git` it uses `$XDG_CONGIF_HOME/git/config`. It's common to not use dot files inside `$XDG_CONFIG_HOME`. `ls $XDG_CONFIG_HOME/*/.* | wc -l` is equal to 1 on my system. and that is from `.yamlfmt` :) When i would go into a dir inside `$XDG_CONFIG_HOME` which contains a dotfile, chances are that i would consider it empty--without noticing the hidden file.

2) You got me wrong, that's my fault. What i meant is. It is often useful to have different configurations for different repos. With `XDG_CONFIG_HOME` it is only possible to have one configuration for all repos. But what if one need an other config in A then B? Then one have to use `.yamlfmt` files in the `$PWD`. Now lets consider this dir struct in a repo:

```
repo
└── src
    ├── a
    │   └── aa.yml
    ├── b
    │   └── bb.yml
    ├── c
    │   └── cc.yml
    └── zz.yml
```

To use `yamlfmt`er one have to put a `.yamlfmt` in all folders (src, src/a, src/b and src/b) but not in `repo`. Most parsers and linters would honor `.yamlfmt` in repo root dir. They sort of searching upwards the `$PWD` dir until they found a `.yamlfmt` file. And if no file is found only then the tool tries to load `$XDG_CONFIG_HOME` and others. I suggest you implement this as well. That's what i meant with spamming `.yamlfmt` files all over.

**Comment by braydonk** [comment]
On `XDG_CONFIG_HOME`:

What you mention re putting the dotfile directly into `$XDG_CONFIG_HOME`, I don't think that's what `yamlfmt` does. It actually does look for `$XDG_CONFIG_HOME/yamlfmt/.yamlfmt`. So `$XDG_CONFIG_HOME/.yamlfmt` shouldn't be a thing and probably shouldn't work. If it does I will try and fix it.

It is a thing in Windows; it defaults to `$HOME/AppData/Local`.

On the search upwards:

Thanks for the explanation it makes more sense now, I think I can make this work in my upcoming revamp of config loading.

**Comment by netzego** [comment]
"What you mention re putting the dotfile directly into $XDG_CONFIG_HOME" no, that's not what i mentioned, do i?  I did not wrote `$XDG_CONFIG_HOME/.yamlfmt`! No additional dot files inside `$XDG_CONFIG_HOME`, as it is already a dotfile. There is no need to "hide" files inside a XDG. No one does it, at least on my system, yamlfmt is the exception here.

**Comment by braydonk** [comment]
Sorry for the misunderstanding. I will be adding support for `yamlfmt.{yml,yaml}` when I address #122 (as well as `.yamlfmt.{yml,yaml}`).

**Comment by netzego** [comment]
#122 makes also sense. So `$XDG_CONFIG_HOME/yamlfmt/yamlfmt.yml` and `$PWD/.yamlfmt.yml`? It's good. Thanks for your time!

**Comment by braydonk** [comment]
The work for this is done and released in v0.11.0
